1password is way to verbose compared to op

## Is your feature request related to a problem? Please describe.
Hi Guys I'm trying to use 1password integration and I'm getting very verbose incatations something like:
```
{{ (onepasswordDetailsFields "qhttpfmbk5vtbfhqtt4oksjyeq").fkegrpik3ol3fta7n7sxsclbsa.value }}
```

Which tells me (in a year) nothing :). While I can get the same data using:
```
{{output "op" "read" "op://_scripts/api_pass.gpgkey/gpgkey"}}
```

I don't see reasons not to use the op command so I will be migrating to this. But I think it would make sense to update the documentation to recommend use of op directly as the onepasswordDetailsFileds plus two meaning less hashes is quite off putting.

It could be done by adding a note to 'Custom' section of the docs  and showing how to use 'op' with the url schema
Some thing like this:
{{secret "read" "op://_scripts/api_pass.gpgkey/gpgkey"}}


I've just learned that `secret` could be used like this while creating this issue.

Oh, just to be clear, I only learnt that the `op://` format existed just the other day so I haven't used it anywhere in my own dotfiles but I might move over to it for my less "secret" tokens 😉

I agree that it's definitely more user friendly and probably would be a reasonable default with the option to use the more advanced GUID path when you need the extra obfuscation.

The only catch with the `op://` syntax is that if you use multiple 1Password accounts (not vaults but actual accounts), you'll need to use the `--account` flag or else chezmoi may not access the correct account.

I think something like this should do it: `{{output "op" "read" "op://Personal/Github/Username", "--account", "my"}}`

```
$ op read op://Personal/Github/Username
[ERROR] 2022/07/24 11:50:06 could not read secret op://Personal/Github/Username: could not get item Personal/Github: "Personal" isn't a vault in this account. Specify the vault with its UUID or name.
$ op read op://Personal/Github/Username --account my
marcus-crane
```

--- a/assets/chezmoi.io/docs/reference/templates/1password-functions/onepasswordRead.md
+++ b/assets/chezmoi.io/docs/reference/templates/1password-functions/onepasswordRead.md
@@ -1,0 +1,24 @@
+# `onepasswordRead` *url* [*account*]
+
+`onepasswordRead` returns data from [1Password](https://1password.com/) using
+the [1Password
+CLI](https://support.1password.com/command-line-getting-started/) (`op`). *url*
+is passed to `op read $URL`. If *account* is specified, the extra arguments
+`--account $ACCOUNT` are passed to `op`.
+
+If there is no valid session in the environment, by default you will be
+interactively prompted to sign in.
+
+!!! example
+
+    The result of
+
+    ```
+    {{ onepasswordRead "op://vault/item/field" }}
+    ```
+
+    is equivalent to calling
+
+    ```console
+    $ op read op://vault/item/field
+    ```

--- a/assets/chezmoi.io/docs/user-guide/password-managers/1password.md
+++ b/assets/chezmoi.io/docs/user-guide/password-managers/1password.md
@@ -27,6 +27,25 @@ This is not necessary if you are using biometric authentication.
     $ eval $(op signin $SUBDOMAIN.1password.com $EMAIL)
     ```
 
+The output of `op read $URL` is available as the `onepasswordRead` template
+function, for example:
+
+```
+{{ onepasswordRead "op://app-prod/db/password" }}
+```
+
+returns the output of
+
+```console
+$ op read op://app-prod/db/password
+```
+
+Documents can be retrieved with:
+
+```
+{{- onepasswordDocument "$UUID" -}}
+```
+
 The output of `op item get $UUID--format json` (`op get item $UUID`) is
 available as the `onepassword` template function. chezmoi parses the JSON output
 and returns it as structured data. For example, if the output is:
@@ -142,12 +161,6 @@ in 1Password have item fields. This can be tested with:
 
 ```console
 $ chezmoi execute-template "{{ onepasswordItemFields \"$UUID\" | toJson }}" | jq .
-```
-
-Documents can be retrieved with:
-
-```
-{{- onepasswordDocument "$UUID" -}}
 ```
 
 !!! note

--- a/assets/chezmoi.io/mkdocs.yml
+++ b/assets/chezmoi.io/mkdocs.yml
@@ -203,6 +203,7 @@ nav:
       - onepasswordDocument: reference/templates/1password-functions/onepasswordDocument.md
       - onepasswordDetailsFields: reference/templates/1password-functions/onepasswordDetailsFields.md
       - onepasswordItemFields: reference/templates/1password-functions/onepasswordItemFields.md
+      - onepasswordRead: reference/templates/1password-functions/onepasswordRead.md
     - AWS Secrets Manager functions:
       - reference/templates/aws-secrets-manager-functions/index.md
       - awsSecretsManager: reference/templates/aws-secrets-manager-functions/awsSecretsManager.md

--- a/pkg/cmd/config.go
+++ b/pkg/cmd/config.go
@@ -462,6 +462,7 @@ func newConfig(options ...configOption) (*Config, error) {
 		"onepasswordDetailsFields": c.onepasswordDetailsFieldsTemplateFunc,
 		"onepasswordDocument":      c.onepasswordDocumentTemplateFunc,
 		"onepasswordItemFields":    c.onepasswordItemFieldsTemplateFunc,
+		"onepasswordRead":          c.onepasswordReadTemplateFunc,
 		"output":                   c.outputTemplateFunc,
 		"pass":                     c.passTemplateFunc,
 		"passFields":               c.passFieldsTemplateFunc,

--- a/pkg/cmd/onepasswordtemplatefuncs.go
+++ b/pkg/cmd/onepasswordtemplatefuncs.go
@@ -358,6 +358,26 @@ func (c *Config) onepasswordOutput(args *onepasswordArgs, withSessionToken withS
 	return output, nil
 }
 
+func (c *Config) onepasswordReadTemplateFunc(url string, args ...string) string {
+	onepasswordArgs := &onepasswordArgs{
+		args: []string{"read", url},
+	}
+	switch len(args) {
+	case 0:
+		// Do nothing.
+	case 1:
+		onepasswordArgs.args = append(onepasswordArgs.args, "--account", args[0])
+	default:
+		panic(fmt.Errorf("expected 1 or 2 arguments, got %d", len(args)))
+	}
+
+	output, err := c.onepasswordOutput(onepasswordArgs, withSessionToken)
+	if err != nil {
+		panic(err)
+	}
+	return string(output)
+}
+
 func (c *Config) onepasswordVersion() (*semver.Version, error) {
 	if c.Onepassword.version != nil || c.Onepassword.versionErr != nil {
 		return c.Onepassword.version, c.Onepassword.versionErr

--- a/pkg/cmd/testdata/scripts/onepassword.txt
+++ b/pkg/cmd/testdata/scripts/onepassword.txt
@@ -21,6 +21,14 @@ stdout '^L8rm1JXJIE1b8YUDWq7h$'
 chezmoi execute-template '{{ (onepasswordItemFields "ExampleLogin").exampleLabel.v }}'
 stdout exampleValue
 
+# test onepasswordRead template function
+chezmoi execute-template '{{ onepasswordRead "op://vault/item/field" }}'
+stdout exampleField
+
+# test onepasswordRead template function with account
+chezmoi execute-template '{{ onepasswordRead "op://vault/item/field" "account" }}'
+stdout exampleAccountField
+
 -- bin/op --
 #!/bin/sh
 
@@ -30,6 +38,12 @@ case "$*" in
     ;;
 "get item ExampleLogin" | "get item ExampleLogin --vault vault --account account" | "--session thisIsAFakeSessionToken get item ExampleLogin" | "--session thisIsAFakeSessionToken get item ExampleLogin --vault vault --account account" | "--session thisIsAFakeSessionToken get item ExampleLogin --account account")
     echo '{"uuid":"wxcplh5udshnonkzg2n4qx262y","templateUuid":"001","trashed":"N","createdAt":"2020-07-28T13:44:57Z","updatedAt":"2020-07-28T14:27:46Z","changerUuid":"VBDXOA4MPVHONK5IIJVKUQGLXM","itemVersion":2,"vaultUuid":"tscpxgi6s7c662jtqn3vmw4n5a","details":{"fields":[{"designation":"username","name":"username","type":"T","value":"exampleuser"},{"designation":"password","name":"password","type":"P","value":"L8rm1JXJIE1b8YUDWq7h"}],"notesPlain":"","passwordHistory":[],"sections":[{"name":"linked items","title":"Related Items"},{"fields":[{"k":"string","n":"D4328E0846D2461E8E455D7A07B93397","t":"exampleLabel","v":"exampleValue"}],"name":"Section_20E0BD380789477D8904F830BFE8A121","title":""}]},"overview":{"URLs":[{"l":"website","u":"https://www.example.com/"}],"ainfo":"exampleuser","pbe":119.083926,"pgrng":true,"ps":100,"tags":[],"title":"ExampleLogin","url":"https://www.example.com/"}}'
+    ;;
+"--session thisIsAFakeSessionToken read op://vault/item/field")
+    echo 'exampleField'
+    ;;
+"--session thisIsAFakeSessionToken read op://vault/item/field --account account")
+    echo 'exampleAccountField'
     ;;
 "signin --raw" | "signin account --raw")
     echo 'thisIsAFakeSessionToken'
@@ -54,6 +68,10 @@ IF "%*" == "--version" (
     echo.{"uuid":"wxcplh5udshnonkzg2n4qx262y","templateUuid":"001","trashed":"N","createdAt":"2020-07-28T13:44:57Z","updatedAt":"2020-07-28T14:27:46Z","changerUuid":"VBDXOA4MPVHONK5IIJVKUQGLXM","itemVersion":2,"vaultUuid":"tscpxgi6s7c662jtqn3vmw4n5a","details":{"fields":[{"designation":"username","name":"username","type":"T","value":"exampleuser"},{"designation":"password","name":"password","type":"P","value":"L8rm1JXJIE1b8YUDWq7h"}],"notesPlain":"","passwordHistory":[],"sections":[{"name":"linked items","title":"Related Items"},{"fields":[{"k":"string","n":"D4328E0846D2461E8E455D7A07B93397","t":"exampleLabel","v":"exampleValue"}],"name":"Section_20E0BD380789477D8904F830BFE8A121","title":""}]},"overview":{"URLs":[{"l":"website","u":"https://www.example.com/"}],"ainfo":"exampleuser","pbe":119.083926,"pgrng":true,"ps":100,"tags":[],"title":"ExampleLogin","url":"https://www.example.com/"}}
 ) ELSE IF "%*" == "--session thisIsAFakeSessionToken get item ExampleLogin --account account" (
     echo.{"uuid":"wxcplh5udshnonkzg2n4qx262y","templateUuid":"001","trashed":"N","createdAt":"2020-07-28T13:44:57Z","updatedAt":"2020-07-28T14:27:46Z","changerUuid":"VBDXOA4MPVHONK5IIJVKUQGLXM","itemVersion":2,"vaultUuid":"tscpxgi6s7c662jtqn3vmw4n5a","details":{"fields":[{"designation":"username","name":"username","type":"T","value":"exampleuser"},{"designation":"password","name":"password","type":"P","value":"L8rm1JXJIE1b8YUDWq7h"}],"notesPlain":"","passwordHistory":[],"sections":[{"name":"linked items","title":"Related Items"},{"fields":[{"k":"string","n":"D4328E0846D2461E8E455D7A07B93397","t":"exampleLabel","v":"exampleValue"}],"name":"Section_20E0BD380789477D8904F830BFE8A121","title":""}]},"overview":{"URLs":[{"l":"website","u":"https://www.example.com/"}],"ainfo":"exampleuser","pbe":119.083926,"pgrng":true,"ps":100,"tags":[],"title":"ExampleLogin","url":"https://www.example.com/"}}
+) ELSE IF "%*" == "--session thisIsAFakeSessionToken read op://vault/item/field" (
+    echo.exampleField
+) ELSE IF "%*" == "--session thisIsAFakeSessionToken read op://vault/item/field --account account" (
+    echo.exampleAccountField
 ) ELSE IF "%*" == "signin --raw" (
     echo thisIsAFakeSessionToken
 ) ELSE IF "%*" == "signin account --raw" (

